BUGFIX: `LinkEditor` placeholder now uses the custom placeholder correctly

dr-format-d opened the pull request:
Fixes #3797


**What I did**
If the user now sets a placeholder for the `LinkEditor` the placeholder will now be correctly shown in the neos frontend.

**How I did it**
I removed a (imo) unnecessary condition, which caused the custom placeholder to be always overridden with a static placeholder.

**How to verify it**
Try to set a placeholder for the `LinkEditor` like you would/expect with any other editor component

**Before:**
![Video before](https://github.com/neos/neos-ui/assets/110388306/c75d004a-7c9c-4a64-bae9-123961a39b91)

**After:**
![Video after](https://github.com/neos/neos-ui/assets/110388306/6edcac58-f091-4194-80f0-111512fef9a0)

dr-format-d committed:
BUGFIX: `LinkEditor` placeholder now uses the custom placeholder correctly

Initially the user could set a placeholder, but inside the `LinkInput.js` it wouldn't be used and an unnecessary static placeholder would be used instead

Fixes #3797

grebaldi commented on packages/neos-ui-editors/src/Library/LinkInput.js:
```diff
@@ -297,7 +297,7 @@ export default class LinkInput extends PureComponent {
                     plainInputMode={isUri(this.state.searchTerm)}
                     onValueChange={this.handleValueChange}
-                    placeholder={this.props.i18nRegistry.translate($get('options.placeholder', this.props) || 'Neos.Neos.Ui:Main:ckeditor__toolbar__link__placeholder', 'Paste a link, or search')}
+                    placeholder={this.props.i18nRegistry.translate($get('options.placeholder', this.props))}
```
```suggestion
                    placeholder={this.props.i18nRegistry.translate($get('options.placeholder', this.props) || 'Neos.Neos.Ui:Main:ckeditor__toolbar__link__placeholder', $get('options.placeholder', this.props) || 'Paste a link, or search')}
```

Otherwise, if no manual placeholder has been configured, the placeholder ends up empty:
![Screenshot_2024-06-03_13-13-22 LinkEditor empty](https://github.com/neos/neos-ui/assets/2522299/9def115f-7cf7-4c60-8826-65a0e519181d)

dr-format-d replied:
Thank you for reviewing! I applied the code you provided, except for the chosen label. I don't see why the label should change from: `Neos.Neos:Main:content.inspector.editors.linkEditor.search` to `Neos.Neos.Ui:Main:ckeditor__toolbar__link__placeholder` only in this case.

Consequently, i adjusted `Paste a link, or search` to `Paste a link, or type to search` to better fit the default placeholder configured here:
<img width="676" alt="image" src="https://github.com/neos/neos-ui/assets/110388306/cfd5dfbd-4a8f-4146-8562-6799c30a047c">

grebaldi replied:
Well, it's not obvious, sadly, but the `<LinkInput/>` is used in two places. One is the LinkEditor (for which the `Neos.Neos:Main:content.inspector.editors.linkEditor.search` label is a fitting choice), the other is the RichTextEditorToolbar, which is why the fallback to `Neos.Neos.Ui:Main:ckeditor__toolbar__link__placeholder` exists.

But, I guess it's okay to unify this and put LinkEditor in the lead. We just gotta remember to clean up the other one :sweat_smile:

grebaldi reviewed (changes requested):
Hi @dr-format-d,

thanks a lot for your first PR to the Neos UI! :tada: 

I tested your change and it makes custom placeholders work correctly. However, I'll need a little adjustment (see comment below), so that it still works, when no custom placeholder is configured.

dr-format-d committed:
BUGFIX: `LinkEditor` placeholder now also displays default placeholder on empty `placeholder:` setting

If someone leaves the `placeholder:` setting empty, the default placeholder, instead of a completely empty `LinkEditor`-input will be shown.

Fixes #3797

Sebobo reviewed (approved):
Thx for this fix! 🙂
